From 26082a96abbf1290d439f3fd17f471bf3f228845 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:01:20 -0400
Subject: [usb] ohci/uhci: add soft dependencies on ehci_pci

Message-id: <1410887463-20674-9-git-send-email-dzickus@redhat.com>
Patchwork-id: 95127
O-Subject: [RHEL7 PATCH 008/591] usb: ohci/uhci - add soft dependencies on ehci_pci
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 05c92da0c52494caf97d58be1b05b1f95a79ce4e
Author: Tom Gundersen <teg@jklm.no>
Date:   Tue Sep 10 23:30:14 2013 +0200

    usb: ohci/uhci - add soft dependencies on ehci_pci

    Support for specifying soft dependencies in the modules themselves was
    introduced in commit 7cb14ba.

    In Arch we have always been shipping a module.d(5) fragment ordering ohci/uhci
    after ehci. If this ordering is really necessary, it would be great to move it
    to the kernel and getting the correct fragment generated by depmod.

    Signed-off-by: Tom Gundersen <teg@jklm.no>
    Cc: Alan Stern <stern@rowland.harvard.edu>
    Cc: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/ohci-pci.c b/drivers/usb/host/ohci-pci.c
index ec337c2..eedf97c 100644
--- a/drivers/usb/host/ohci-pci.c
+++ b/drivers/usb/host/ohci-pci.c
@@ -323,3 +323,4 @@ module_exit(ohci_pci_cleanup);
 
 MODULE_DESCRIPTION(DRIVER_DESC);
 MODULE_LICENSE("GPL");
+MODULE_SOFTDEP("pre: ehci_pci");
diff --git a/drivers/usb/host/uhci-pci.c b/drivers/usb/host/uhci-pci.c
index 0f228c4..d89c78d 100644
--- a/drivers/usb/host/uhci-pci.c
+++ b/drivers/usb/host/uhci-pci.c
@@ -299,3 +299,5 @@ static struct pci_driver uhci_pci_driver = {
  },
 #endif
 };
+
+MODULE_SOFTDEP("pre: ehci_pci");
-- 
1.7.1